src/pcapi/core/users/repository.py

```diff
@@ -47,20 +48,21 @@ def get_user_with_credentials(identifier: str, password: str) -> User:
 
 
 def get_user_with_valid_token(
-    token_value: str, token_types: list[models.TokenType], delete_token: bool = False
+    token_value: str, token_types: list[models.TokenType], use_token: bool = True
 ) -> Optional[User]:
     token = models.Token.query.filter(
-        models.Token.value == token_value, models.Token.type.in_(token_types)
+        models.Token.value == token_value, models.Token.type.in_(token_types), models.Token.isUsed == False
     ).one_or_none()
     if not token:
         return None
 
     if token.expirationDate and token.expirationDate < datetime.now():
         return None
 
-    if delete_token:
-        models.Token.query.filter_by(id=token.id).delete()
+    if use_token:
+        token.isUsed = True
 
+    repository.save(token)
```

Comment:
```suggestion
        repository.save(token)
```
Il faudrait regarder dans le flux global que ça ne régénère pas plein de requêtes (on aura au moins celle du token + user).